### PR TITLE
Feature/0.2.2.15/tidy code

### PR DESCRIPTION
**Pull request by sharrief**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

- Rename of ClientService to DataService and export functions instead of singleton instance
- Clean up of unused code from ClientService/AnimalService. Either deleted or moved into DataService
- Clean up code from /client path
- Create DataService module to hold data layer queries

## Related Tickets & Documents

https://projects.digitalaidseattle.org/project-tasks-details?recordId=recWhsp2ZkCR7KlpF

- Related Issue # 0.2.2.15 Tidy code
- Closes # 0.2.2.15 Tidy code

## QA Instructions, Screenshots, Recordings

N/A

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Color contrast tested?

## Added/updated tests?
_We are aiming for 100% statement coverage._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] Which Seattle Humane animal would you like to dedicate this PR to?

Max
Max

### CODE REVIEW

File: __tests__/src/services/DataService.test.ts [hunk 1]
```diff
@@ -112,46 +113,76 @@ describe('ClientService', () => {
...
     });
   });
-  describe('static getRecentTickets()', () => {
+  describe('static getServiceRequestSummary()', () => {
     it('returns recent tickets from the db', async () => {
       // Arrange
       const expectedTickets = recentTickets;
-      mockSupabaseClient.setTestData(expectedTickets);
+      /**
+       * TODO research a better way to mock consecutive queries to supabase client
```
Comment (DillonOLeary) [inline __tests__/src/services/DataService.test.ts:124]: I was running into the same problem! I'm not sure

File: src/hooks/__mocks__/useTeamMembers.ts [hunk 2]
```diff
@@ -1,6 +1,25 @@
-export const john = { id: 'j456', value: 'abc', label: 'john@test.digitalseattle.org' };
-export const jane = { id: 'j123', value: 'xyz', label: 'jane@test.digitalseattle.org' };
-export const teamMembers = [jane, john];
-export default function useTeamMembers() {
-  return teamMembers;
+import { faker } from '@faker-js/faker';
+import type { TeamMeberOption } from '@hooks/useTeamMembers';
+import type { TeamMemberType } from '@types';
+
+faker.seed(123);
```
Comment (DillonOLeary) [inline src/hooks/__mocks__/useTeamMembers.ts:5]: oh wow cool!

File: __tests__/src/components/serviceRequest/ServiceRequestDialog.test.tsx
Comment (DillonOLeary) [inline __tests__/src/components/serviceRequest/ServiceRequestDialog.test.tsx:1]: I'm not sure I fully understand the changes to this file. At a high level, what did you do here?

File: src/components/ClientDialog.tsx [hunk 3]
```diff
@@ -77,18 +77,7 @@ function ClientDialog(props: ClientDialogProps) {
...
-      ...request,
-      // TODO not sure how we want to handle these ids, needs lookup control?
-      pet_id: null,
-      team_member_id: null,
-    }, client, animal)
-      .then((requestResponse) => props.onClose(requestResponse))
-      // TODO - handle all sorts of errors: client exists, animal exists, request exists, etc.
-      .catch((err) => props.onClose(null))
-      .finally(() => setBusy(false));
+    // TODO implement client creation
```
Comment (DillonOLeary) [inline src/components/ClientDialog.tsx:80]: how does getting rid of the client creation affect the code? Is it handled somewhere else?

**Review by DillonOLeary — COMMENTED** [review]
I left some comments for clarification. I think I understand generally what you are doing, but got lost in some of the details. One thing that might be tricky though is the PRs that Adele plans to merge, you may need to do some merge work.

But it looks like you put a lot of time and thought into this, thanks Sharrief!

File: __tests__/src/components/serviceRequest/ServiceRequestDialog.test.tsx
Comment (DillonOLeary) [reply __tests__/src/components/serviceRequest/ServiceRequestDialog.test.tsx:1]: I took another look it makes more sense. It seems like it was a part of the refactor to use data service. Plus a more robust test for ticket creation.

File: src/components/ClientDialog.tsx (hunk 3 above)
Comment (sharrief) [reply src/components/ClientDialog.tsx:80]: This code wasn't creating a client. It was creating a service request. It was code that was copied over from another place that has already been cleaned up. 

We  need to add the behavior for creating a client. The good thing is we now have the queries and the UI components for it (the client section), we just need to wire them up!

**Review by DillonOLeary — DISMISSED** [review]
Looks good now! You might want to coordinate with Adele on his merges though, could complicate things

**Review by DillonOLeary — APPROVED** [review]

**Pull request merged**